### PR TITLE
MSMS fragmentation match

### PR DESCRIPTION
**Pull request by shubhra-agrawal**
This PR implements the MSMS spectral matching feature. This feature has been reimplemented from upstream MAVEN. Following changes have been included:

- Migrate and clean up Fragment class (and related members in other classes like PeakGroup, Compound)
- Migrate and clean up code for calculating Fragmentation pattern of every group during automated detection if "Match Fragmentation" is on
- Match the group and compound spectra to assign match scores based on multiple algorithms
- Filter out groups with a lower Hypergeometric MS2 score than the "minFragMatchScore"

### CODE REVIEW

File: src/core/libmaven/EIC.h [hunk 1]
```diff
@@ -222,7 +222,7 @@ class EIC
     void subtractBaseLine();
     void clearEICContents();
     void interpolate();
-    vector<Scan *> getFragmenationEvents();
+    vector<Scan *> getFragmentationEvents();
```
[the remark this answers is not in the document]
Comment (shubhra-agrawal) [reply src/core/libmaven/EIC.h]: It's not being used anywhere in the later PRs either, but let's keep it for future use.

[the remark this answers is not in the document]
Comment (shubhra-agrawal) [reply src/core/libmaven/EIC.h]: I take that back. This function is a duplicate of the one in mzSample so I'll remove it.

File: src/core/libmaven/mzSample.cpp [hunk 2]
```diff
@@ -1819,6 +1819,20 @@ void mzSample::restorePreviousRetentionTimes()
 	}
 }
 
+vector<Scan*> mzSample::getFragmentationEvents(mzSlice* slice)
```
[the remark this answers is not in the document]
Comment (shubhra-agrawal) [reply src/core/libmaven/mzSample.cpp]: I find that having the same function name across classes for a particular functionality helps. `getFragmentationEvents()` in PeakGroup is just acting as a wrapper for the one in mzSample, not performing a different function. 

If the function name itself is not clear enough, we can change that to `getMS2Scans()` for both. What do you think?

**Comment by shubhra-agrawal** [comment]
@rish9511 @saifulbkhan Will make sure to add documentation once the development is over for the whole feature.

File: src/core/libmaven/mzSample.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (rish9511) [reply src/core/libmaven/mzSample.cpp]: Lets stick to `getFragmentationEvents()`

**Pull request merged**